### PR TITLE
Add pad_to_aspect_ratio arg to image_dataset_from_directory

### PR DESCRIPTION
**Pull request by lukasbank47**
I'm not really sure how to do this but I've noticed that there is no easy way to use padding with image_dataset_from_directory. So I added an analogous argument to crop_to_aspect_ratio. I'm not sure about the behaviour when both padding and croping are set to true so I just raise an error.

### CODE REVIEW

**Review by fchollet — APPROVED** [review]
That's a useful feature, thanks for the PR!

File: keras/utils/image_dataset_utils.py [hunk 1]
```diff
@@ -384,13 +397,30 @@ def load_image(
...
-    if crop_to_aspect_ratio:
+    
+    if pad_to_aspect_ratio and crop_to_aspect_ratio:
+        raise ValueError(
+                'Only one of pad_to_aspect_ratio, crop_to_aspect_ratio can be set to True'
+            )
+    
+    if pad_to_aspect_ratio:
+        if data_format == "channels_first":
+            img = tf.transpose(img, (2, 0, 1))
```
Comment (fchollet) [inline keras/utils/image_dataset_utils.py:415]: This should be move out of the `if` block to reduce code redundancy.

Also the current logic seems not to support channels_first, please fix: https://github.com/keras-team/keras/actions/runs/8170560185/job/22337254139?pr=19258

**Review by fchollet — COMMENTED** [review]
To format the code, use `sh shell/format.sh`.

**Comment by fchollet** [comment]
Almost there! Please take a look at the failing tests.

File: keras/utils/image_dataset_utils.py [hunk 2]
```diff
@@ -410,25 +410,28 @@ def load_image(
...
 
-    if data_format == "channels_first":
-        img = tf.transpose(img, (2, 0, 1))
-
-    if pad_to_aspect_ratio:
-        img = tf.image.resize_with_pad(
-            img, image_size[0], image_size[1], method=interpolation
-        )
-    elif crop_to_aspect_ratio:
+    if crop_to_aspect_ratio:
```
Comment (lukasbank47) [inline keras/utils/image_dataset_utils.py:413]: Change the order of transpose to behind the pad operation. I though about extracting the "channels_first" case but I think this is clearer

**Review by fchollet — APPROVED** [review]
LGTM, thanks!

**Pull request merged**